piptools/scripts/compile.py

```diff
@@ -442,6 +444,14 @@ def cli(
                     for req in metadata.get_all("Requires-Dist") or []
                 ]
             )
+            # Since "*" is not a valid extra there is no risk confusing an actual extra
+            # for the wildcard.
+            # https://peps.python.org/pep-0508/#grammar
+            if "*" in extras:
```

Comment:
What do you think, shouldn't we use an explicit option (e.g., `--all-extras`) to avoid making users think that a wildcard can be used as a regexp like `--extras=foo*`? Also, as commented, "'*' is not a valid extra".

Reply:
Good point about the pattern patching, will fix.

Reply:
Awesome, thanks!